chore(example): Updated example app to 0.61.5

Naturalclar opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please follow the template so that the reviewers can easily understand what the code changes affect -->

# Summary

<!--
Explain the **motivation** for making this change: here are some points to help you:

* What issues does the pull request solve? Please tag them so that they will get automatically closed once the PR is merged
* What is the feature? (if applicable)
* How did you implement the solution?
* What areas of the library does it impact?
-->
Upgraded the react-native version of example folder to RN 0.61.5

## Test Plan

| iOS | Android |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/6936373/71556196-812eee00-2a78-11ea-939c-fdc81f2914c0.png" width="320" /> | <img src="https://user-images.githubusercontent.com/6936373/71556195-812eee00-2a78-11ea-9aa7-4a9f3d089a6a.png" width="320" /> |


<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

### What's required for testing (prerequisites)?

### What are the steps to reproduce (after prerequisites)?

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

<!-- Check completed item, when applicable, via: [X] -->

- [x] I have tested this on a device and a simulator
- [x] I added a sample use of the API in the example project (`example/App.js`)

jsunwoo commented:
Hi,

I'm trying to implement react-native-voice to react-native 0.60.2
The package seems doesn't support above 0.60 versions.

Even though it was hard, I kept trying it and found your pull request.
You seem like succeeded to implement the package to 0.61.5
Your demo runs perfectly.

I'd like to know how you migrate the package to above the 0.61.5 version.
After installing the npm package, which process I need to do to migrate?

Any helps would be grateful.
Thank you in advance.

Naturalclar commented:
@jsunwoo 
Starting from react-native 0.60+, you will need to run `cd ios && pod install`
after installing the package for iOS projects.
Also, if you were previously using older version of react-native (<= 0.59), you may need to first unlink the package by running `react-native unlink react-native-voice`

jsunwoo commented:
@Naturalclar

Thank you for your reply.
After installing the package and I ran `cd ios && pod install`
And run with `react-native run-ios` (also I unlinked the package)


<img src="https://user-images.githubusercontent.com/18051496/73130645-c88ca800-403f-11ea-9819-4618f62859a8.png" width="200">


Then, it runs well like the screenshot above.
When I click the record button, the app crushed like this below.

<img src="https://user-images.githubusercontent.com/18051496/73130647-d5110080-403f-11ea-939f-969878b6ef02.png" width="200">

What am I missing now?
Would you give me some suggestions?

jsunwoo commented:
This is my app's package.json

`
{
  "name": "aweproject",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "react": "16.9.0",
    "react-native": "0.61.2",
    "react-native-voice": "^0.3.0",
    "react-native-youtube": "^2.0.0"
  },
  "devDependencies": {
    "@babel/core": "^7.8.3",
    "@babel/runtime": "^7.8.3",
    "@react-native-community/eslint-config": "^0.0.6",
    "babel-jest": "^25.1.0",
    "eslint": "^6.8.0",
    "jest": "^25.1.0",
    "metro-react-native-babel-preset": "^0.58.0",
    "react-test-renderer": "16.9.0"
  },
  "jest": {
    "preset": "react-native"
  }
}
`

Naturalclar commented:
@jsunwoo 
You may need to add an appropriate permission to your `info.plist` to use voice recording feature. Could you make sure that strings below are included in your `info.plist`?
```
<dict>
  ...
  <key>NSMicrophoneUsageDescription</key>
  <string>Description of why you require the use of the microphone</string>
  <key>NSSpeechRecognitionUsageDescription</key>
  <string>Description of why you require the use of the speech recognition</string>
  ...
</dict>
```

jsunwoo commented:
<img src="https://user-images.githubusercontent.com/18051496/73130784-202c1300-4042-11ea-95f4-e7be9df8b9b9.png" width="200">

It works beautifully.
Thank you @Naturalclar 🙏

lucasbento commented:
Hey @Naturalclar, awesome work.

Is there anything left to do here or this just needs a review?

Naturalclar commented:
@lucasbento should be good to go! :+1:

lucasbento commented:
cc @safaiyeh, can you check this when you have some time? 😊

safaiyeh commented:
Sounds good I’ll check this out tomorrow!

safaiyeh commented on VoiceTest/android/.project:
```diff
@@ -0,0 +1,17 @@
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+</projectDescription>
```
Should be apart of the gitignore

safaiyeh commented on VoiceTest/android/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
+eclipse.preferences.version=1
```
Same with this

safaiyeh commented on VoiceTest/package.json:
```diff
@@ -1,24 +1,27 @@
+    "babel-jest": "^24.9.0",
+    "jest": "^24.9.0",
+    "metro-react-native-babel-preset": "^0.56.0",
+    "react-test-renderer": "16.9.0"
```
package.json should be shared with the root, so we can easily run the example.

Naturalclar replied:
I'm trying to get the example to work with a single `package.json` from the root, but i'm unable to run them. :thinking:
running `npx react-native start` will look for an `index.js` in the root directory, not in the `example` folder. is there a way to make them work with a single package.json in the root?

Naturalclar replied:
never mind, i saw the examples in `react-native-webview` which uses --root option for the cli. let me try with that :+1:

Naturalclar replied:
I've tried it with the way `react-native-webview` does it, but `react-native start` stills tries to find an `index.js` in the root directory. it works when I add an `index.js` in the root directory, but I'm not sure if that is what we want (we don't really want to ditribute an extra content that is only used in the `example` folder)
I'm gonna give it another try tomorrow!

safaiyeh replied:
Hmm maybe we can move the index file from `/src` to the root? There is nothing else in the folder. What do you think about that?

safaiyeh replied:
@Naturalclar

lucasbento replied:
Just chipping in here, do you really need to use the same `package.json`? You can keep them separate and have a script on the root-level one to run the example like `react-navigation` does: https://github.com/react-navigation/react-navigation/blob/master/package.json#L23

Naturalclar replied:
@safaiyeh I agree with @lucasbento that examples can have its own `package.json`.
I think the example folder should serve as a reference for users that wants to use the package, and having a `package.json` to know how its structured might be helpful to the users. Also, we can keep unrelated devDependencies just for examples out of the root `package.json`.
I've made some scripts that can run examples for the root `package.json`, like how its done in react-navigation. How do you feel? (That and I'm having a little hard time getting the single package.json to work :sob:)

safaiyeh replied:
Okay no worries since it has been a pain! @Naturalclar approved ✅

safaiyeh commented:
Also could we name the folder `example`.

Naturalclar commented:
@safaiyeh 
renamed the folder to `example` and removed the unnecessary packages for android and put them in `.gitignore` :+1:
I'm having trouble removing the package.json to have a single `package.json` in the root. could you take a look at my comment?

safaiyeh commented:
Checkout webview (https://github.com/react-native-community/react-native-webview), we have it working with a single package.json @Naturalclar

safaiyeh commented:
If you're not aware of semantic syntax, merge it with the message `chore(example): Updated example app` something like that! @Naturalclar

Naturalclar commented:
@safaiyeh got it :+1: thanks for the info about the semantic syntax!
I'll merge it once CI passes.
